### ISSUE TITLE
Wrong line breaking on newline for bash-4.2

### ISSUE THREAD

**Issue by doomhz** [docker/for-mac#1681]
### Expected behavior

Proper line breaking on newlines.

### Actual behavior

Wrong rendering on bash-4.2:
![example](https://i.imgur.com/dOAckK4.png)

### Information

Dist `4.9.30-moby`, `Amazon Linux AMI release 2016.03`

```
Docker for Mac: version: 17.06.0-rc1-ce-mac13 (4cdec4294a50b2233146b09469b49937dabdebdd)
macOS: version 10.12.5 (build: 16F73)
logs: /tmp/2FAB74C8-3C32-4ED5-95D1-A0A65B9491A6/20170602-184144.tar.gz
[OK]     db.git
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     db
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     slirp
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[OK]     docker-cli
[OK]     menubar
[OK]     disk
```

**Comment by omata-metamax** [comment]
I have the same issue just after upgrade my docker version

## Docker Info:

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 17.06.0-ce-rc1
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 3addd840653146c90a254301d6c3a663c7fd6429
runc version: 992a5be178a62e026f4069f443c6164912adbf09
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.30-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.952GiB
Name: moby
ID: ZY2T:EU6H:BLYL:SYAK:2ZWR:ZPQM:76KR:SLFK:YPJM:QAZM:5R2P:IDEN
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 18
 Goroutines: 30
 System Time: 2017-06-02T16:58:32.950889395Z
 EventsListeners: 1
No Proxy: *.local, 169.254/16
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

## Docker Version:

```
Client:
 Version:      17.06.0-ce-rc1
 API version:  1.30
 Go version:   go1.8.1
 Git commit:   7f8486a
 Built:        Wed May 31 02:56:01 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      17.06.0-ce-rc1
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.1
 Git commit:   7f8486a
 Built:        Wed May 31 03:00:14 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

**BTW:** The OS is wrongly reported by Docker.

# Real OS info:

## uname -a

```
Darwin 16.6.0 Darwin Kernel Version 16.6.0: Fri Apr 14 16:21:16 PDT 2017; root:xnu-3789.60.24~6/RELEASE_X86_64 x86_64
```

## sw_vers -productVersion

```
10.12.5
```

## system_profiler SPSoftwareDataType

```
Software:

    System Software Overview:

      System Version: macOS 10.12.5 (16F73)
      Kernel Version: Darwin 16.6.0
      Boot Mode: Normal
      Secure Virtual Memory: Enabled
      System Integrity Protection: Enabled
```

**Comment by avirshup** [comment]
Same behavior and environment here, just wanted to add that this began instantly after upgrading to 17.06.0-rc1-ce-mac13

**Comment by charles-dyfis-net** [comment]
Running `stty onlcr` appears to be an effective workaround.

**Comment by mkubenka** [comment]
Duplicate #1672

**Comment by kinghuang** [comment]
Looks like this is fixed by moby/moby#33502.
